misk/src/main/kotlin/misk/MiskApplication.kt

```diff
@@ -88,6 +90,10 @@ class MiskApplication(private val modules: List<Module>, commands: List<MiskComm
     Runtime.getRuntime().addShutdownHook(object : Thread() {
       override fun run() {
         log.info { "received a shutdown hook! performing an orderly shutdown" }
+        if (injector.getExistingBinding(Key.get(WebConfig::class.java)) != null) {
+          val config = injector.getInstance<WebConfig>()
+          sleep(config.shutdown_sleep_ms.toLong())
```

Comment:
This is my attempt to make the delay configurable. It's not clear to me what other avenues exist for making it possible for consumers to customize behaviour at this layer of the framework.